specification/configuration.md

```diff
@@ -89,6 +90,50 @@ environment variables:
   If `SPLUNK_CONFIG` is defined then these options are not required but MAY be
   defined.
 
+### Kubernetes
+
+While Kubernetes supports container technology that can be configured using
+environment variables, package management solutions such as Helm charts and
+Operators require YAML-based configuration. As a result, Kubernetes package
+management solutions MUST support the following YAML configuration options:
+
+- `splunkCommon`
+  - `clusterName` ()          : [REQUIRED] Name of the cluster.
+  - `distro` ()               : Which distribution of Kubernetes is deployed.
+  - `environment` ()          : Name of the environment; if not defined then skipped.
+  - `logsEngine` (`fluentd`)  : How to collect logs.
+  - `modes`
+    - `agent`                 : Deployed as a DaemonSet.
+      - `config` ()           : Updates configuration. Non-list options merged, list options override.
+    - `k8sClusterReceiver`    : Deployed as a standalone service.
+      - `config` ()           : Updates configuration. Non-list options merged, list options override.
```

Comment:
Both of them need `enabled` option as well. For example, they both can be disabled in order to deploy gateway-only mode

Reply:
Ah, makes sense.